integration/setup/tenant_cluster.go

```diff
@@ -87,14 +92,57 @@ func EnsureTenantClusterDeleted(ctx context.Context, id string, config Config, w
 	return nil
 }
 
-func crExistsCondition(ctx context.Context, id string, config Config) release.ConditionFunc {
+func crExistsCondition(ctx context.Context, config Config, crd *apiextensionsv1beta1.CustomResourceDefinition, crNamespace, crName string) release.ConditionFunc {
 	return func() error {
-		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waiting for creation of CR %#q in namespace %#q", id, crNamespace))
+		// In client-go@10.12.x it will be:
+		//
+		//	gvr := metav1.GroupVersionResource{
+		//		Group:    crd.Spec.Group,
+		//		Version:  crd.Spec.Version,
+		//		Resource: crd.Spec.Names.Plural,
+		//	}
+		//
+		resource := &metav1.APIResource{
+			Name:       crd.Spec.Names.Plural,
+			Namespaced: crd.Spec.Scope == "Namespaced",
+		}
+		gv := &schema.GroupVersion{
+			Group:   crd.Spec.Group,
+			Version: crd.Spec.Version,
+		}
+
+		var dynamicClient *dynamic.Client
+		{
+			var err error
+
+			c := config.Host.RestConfig()
+			configShallowCopy := *c
+			configShallowCopy.APIPath = "/apis"
+			configShallowCopy.GroupVersion = gv
+			if configShallowCopy.UserAgent == "" {
+				configShallowCopy.UserAgent = rest.DefaultKubernetesUserAgent()
+			}
```

Comment:
I expect that to be a little bit less cryptic in `client-go@10.12.x` but I don't know yet.